CDTDatastore/query/CDTQIndex.m

```diff
@@ -82,6 +86,23 @@ + (instancetype)index:(NSString *)indexName
                ofType:(NSString *)indexType
          withSettings:(NSDictionary *)indexSettings
 {
+    CDTQIndexType typeAsEnum = [CDTQIndexManager indexTypeForString:indexType];
+    return [[self class] index:indexName withFields:fieldNames type:typeAsEnum withSettings:indexSettings];
+}
+
+#pragma mark New creator methods
+
+
++ (instancetype)index:(NSString *)indexName withFields:(NSArray<NSString *> *)fields type:(CDTQIndexType)type
+{
+        return [[self class] index:indexName withFields:fields type:type withSettings:nil];
+}
+
++ (instancetype)index:(NSString *)indexName
+           withFields:(NSArray *)fieldNames
+                 type:(CDTQIndexType)indexType
+         withSettings:(NSDictionary *)indexSettings
```

Comment:
There's a bit of inconsistency in the use of no, `with` and `of` as prefixes to parts of method names. Can this be improved or are we hamstrung by existing method name?